### PR TITLE
Cleanup of unused code in Identity

### PR DESCRIPTION
**Pull request by HenrikWM**
Summary of the changes within `UserManager.cs`:
- Removed unused code
- Added `readonly` to a private field
- Renamed local variable from `store` to `claimStore`

### CODE REVIEW

**Comment by HenrikWM** [comment]
I’ll fix the tests in a few days.

**Comment by HenrikWM** [comment]
Tests are green ✔️

File: src/Identity/test/Identity.Test/UserManagerTest.cs [hunk 1]
```diff
@@ -657,9 +657,9 @@ public async Task ClaimMethodsFailWhenStoreNotImplemented()
         {
             var manager = MockHelpers.TestUserManager(new NoopUserStore());
             Assert.False(manager.SupportsUserClaim);
-            await Assert.ThrowsAsync<NotSupportedException>(async () => await manager.AddClaimAsync(null, null));
+            await Assert.ThrowsAsync<NotSupportedException>(async () => await manager.AddClaimsAsync(null, null));
```
Comment (HaoK) [inline src/Identity/test/Identity.Test/UserManagerTest.cs]: Removing the GetClaimStores is what caused the regression as those are what throw the not supported exceptions

Comment (HenrikWM) [reply src/Identity/test/Identity.Test/UserManagerTest.cs]: Yes but the store wasn't really being used in `AddClaimAsync` or `RemoveClaimAsync`. Both are just calling `GetClaimStore` without the claim store being used for anything. It's in `AddClaimsAsync` and `RemoveClaimsAsync` that the store is really used. These two methods are just passing parameters along to the overloads. So we're not changing any behavior, just throwing closing to usage than we were before.

So I've changed the methods to assert for the `NotSupportedException` on the actual methods using the store.

**Comment by HaoK** [comment]
This is a regression, as we rely on the Get checks to throw the exceptions

**Pull request closed without merging**